schemas/skills/samples/complex-skillmanifest-2.1.preview-0.json

```diff
@@ -0,0 +1,184 @@
+            "bookFlight": "#/activities/bookFlight",
+            "getWeather": "#/activities/getWeather",
+            "*": "#/activities/message"
+        }
+    },
+    "activities": {
+        "bookFlight": {
+            "description": "Books a flight",
+            "type": "event",
+            "name": "BookFlight",
+            "value": {
+                "$ref": "#/definitions/bookingInfo"
+            },
+            "resultValue": {
+                "$ref": "#/definitions/bookingInfo"
+            }
+        },
+        "getWeather": {
+            "description": "Retrieves and returns the weather for the user's location",
+            "type": "invoke",
+            "name": "GetWeather",
+            "value": {
+                "$ref": "#/definitions/location"
+            },
+            "resultValue": {
+                "$ref": "#/definitions/weatherReport"
+            }
+        },
+        "message": {
+            "type": "message",
+            "description": "Receives the user's' utterance and attempts to resolve it using the skill's LU models"
+        }
+    },
+    "definitions": {
+        "localeValue": {
+            "type": "object",
+            "properties": {
+                "locale": {
+                    "type": "string",
+                    "description": "The current user's locale ISO code"
+                }
+            }
+        },
+        "bookingInfo": {
+            "type": "object",
+            "required": [
+                "origin"
+            ],
+            "properties": {
+                "origin": {
+                    "type": "string",
+                    "description": "this is the origin city for the flight"
+                },
+                "destination": {
+                    "type": "string",
+                    "description": "this is the destination city for the flight"
+                },
+                "date": {
+                    "type": "date",
+                    "description": "The date for the flight in YYYY-MM-DD format"
+                }
+            }
+        },
+        "weatherReport": {
+            "type": "array",
+            "description": "Array of forecasts for the next week.",
+            "items": [
+                {
+                    "type": "string"
+                }
+            ]
+        },
+        "location": {
+            "type": "object",
+            "description": "Location metadata",
+            "properties": {
+                "latitude": {
+                    "type": "number",
+                    "title": "Latitude"
+                },
+                "longitude": {
+                    "type": "number",
+                    "title": "Longitude"
+                },
+                "postalCode": {
+                    "type": "string",
+                    "title": "Postal code"
+                }
+            }
+        }
+    },
+    "activitiesSent": {
+        "flightUpdated": {
+            "type": "event",
```

Comment:
as above - are events the proposed way forward or invoke?

Reply:
The most common one will be event (and the one I will personally mostly recommend), Teams uses invokes in some situations but they are very uncommon.